x/cronos/keeper/precompiles/utils.go

```diff
@@ -0,0 +1,50 @@
+package precompiles
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/ethermint/x/evm/keeper/precompiles"
+)
+
+type NativeMessage interface {
+	codec.ProtoMarshaler
+	GetSigners() []sdk.AccAddress
+}
+
+// exec is a generic function that executes the given action in statedb, and marshal/unmarshal the input/output
+func exec[Req NativeMessage, Resp codec.ProtoMarshaler](
```

Comment:
I didnt know can infer Req from function param when init type👍

Reply:
I'm not sure if it works though.

Reply:
nil pointer fixed.